frappe/model/base_document.py

```diff
@@ -669,8 +669,9 @@ def db_insert(self, ignore_if_duplicate=False):
 					raise frappe.DuplicateEntryError(self.doctype, self.name, e)
 
 			elif frappe.db.is_unique_key_violation(e):
-				# unique constraint
-				self.show_unique_validation_message(e)
+				if not ignore_if_duplicate:
```

Comment:
Current behaviour is intentional, that's why we bother differentiating `PRIMARY` vs rest in db code. 

Duplicate record and unique key violations are different things. This change is breaking IMO.

Reply:
https://github.com/frappe/frappe/blob/0a907ccb4cf18c110a0ca3d19f746486a5998b6c/frappe/utils/install.py#L93C37-L93C43 so this was causing an issue with sqlite, since it gets created earlier when importing report doctype json.

Will just drop that line instead and see.